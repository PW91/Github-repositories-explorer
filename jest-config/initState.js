export default {
  ui: {
    activeUserCard: null,
    lastSearchQuery: null,
    isSearchEmpty: null,
    isLoading: null
  },
  users: []
}