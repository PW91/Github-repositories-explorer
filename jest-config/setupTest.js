import Enzyme from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import 'regenerator-runtime/runtime.js'
import '~/i18n'

Enzyme.configure({
  adapter: new EnzymeAdapter(),
  disableLifecycleMethods: true
})
