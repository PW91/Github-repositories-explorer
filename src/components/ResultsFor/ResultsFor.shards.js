import styled from '@emotion/styled'
import { Paragraph } from '~/components/Typography'
import { IcoClose } from '~/styles/icofont'

export const ResultsForContainer = styled.div`
  display: flex;
  align-items: center;
  margin: 1rem 0;

  ${Paragraph} {
    margin-right: .5rem;
  }
`

export const ClearIcon = styled(IcoClose)`
  margin-left: 1rem;
  cursor: pointer;
  font-size: ${({ theme }) => theme.fontSize.icon.normal};
  color: ${({ theme }) => theme.colors.text.main};
`
