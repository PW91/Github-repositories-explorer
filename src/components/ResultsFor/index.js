export { default } from '~/components/ResultsFor/ResultsForView'
