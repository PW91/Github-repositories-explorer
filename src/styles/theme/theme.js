import * as colors from '~/styles/theme/colors'
import * as commons from '~/styles/theme/commons'

export default {
  colors,
  ...commons
}
