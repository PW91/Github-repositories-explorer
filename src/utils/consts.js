export const API_URL = 'https://api.github.com'

export const BREAKPOINTS = {
  SM: 480,
  MD: 768,
  LG: 992,
  XL: 1300,
}

export const SEARCH_RESULTS_LIMIT = 5

export const API_ACTIONS = {
  USERS_SEARCH: 'USERS_SEARCH',
  USER_REPOS: 'USER_REPOS'
}

export const STORE_ACTIONS = {
  SET_ACTIVE_USER_CARD: 'SET_ACTIVE_USER_CARD',
  SET_LAST_SEARCH_QUERY: 'SET_LAST_SEARCH_QUERY',
  SET_SEARCH_EMPTY_FLAG: 'SET_SEARCH_EMPTY_FLAG',
  SET_LOADING_FLAG: 'SET_LOADING_FLAG',
  SET_USERS: 'SET_USERS',
  SET_USER_REPOS: 'SET_USER_REPOS'
}
