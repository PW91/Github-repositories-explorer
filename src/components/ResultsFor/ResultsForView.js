import React from 'react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import { Paragraph, NormalHeading } from '~/components/Typography'
import { ResultsForContainer, ClearIcon } from '~/components/ResultsFor/ResultsFor.shards'

const ResultsForView = ({ searchQuery, clearingFn }) => {
  const { t } = useTranslation('common')
  return (
    <ResultsForContainer data-test='ResultsForView'>
      <Paragraph>{t('resultsFor')}</Paragraph>
      <NormalHeading>{searchQuery}</NormalHeading>
      <ClearIcon onClick={clearingFn} />
    </ResultsForContainer>
  )
}

ResultsForView.propTypes = {
  clearingFn: PropTypes.func,
  searchQuery: PropTypes.string
}

export default ResultsForView
