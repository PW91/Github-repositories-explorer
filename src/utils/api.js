import { format } from 'url'
import { toast } from 'react-toastify'
import { parseJSON } from '~/utils/helpers'
import { API_URL, API_ACTIONS } from '~/utils/consts'

const ajaxer = (apiAction, path, method = 'GET', data = null, query = null) => (
  new Promise((resolve, reject) => {
    const sendForm = data && data.constructor === FormData
    const requestOptions = {
      headers: !sendForm && { 'content-type': 'application/json' },
      method,
      ...!!data && { body: sendForm ? data : JSON.stringify(data) },
    }

    const fullPath = `${API_URL}${path}${query ? format({ query }) : ''}`

    fetch(fullPath, requestOptions)
      .then(res => res.status < 400 ? res.text() : { errorText: res.statusText, status: res.status })
      .then(res => {
        if (res.errorText) {
          const { errorText, status } = res
          toast.error(errorText)
          reject({ errorText, status })
        } else {
          const parsedJson = parseJSON(res)
          const mapped = responseFormatter(parsedJson, apiAction)
          resolve(mapped)
        }
      })
  })
)

export const responseFormatter = (data, apiAction) => {
  switch (apiAction) {
    case API_ACTIONS.USERS_SEARCH:
      return data.items.map(({
        login,
        id,
        avatar_url
      }) => ({
        login,
        id,
        avatarUrl: avatar_url
      }))
    case API_ACTIONS.USER_REPOS:
      return data.map(({
        id,
        owner,
        name,
        html_url,
        stargazers_count,
        description
      }) => ({
        id,
        owner: owner.id,
        name,
        htmlUrl: html_url,
        stars: stargazers_count,
        description
      }))
  }
}

export const manageUsersSearch = {
  get: (query) => ajaxer(API_ACTIONS.USERS_SEARCH, '/search/users', 'GET', null, query)
}

export const manageUserRepos = {
  get: (username) => ajaxer(API_ACTIONS.USER_REPOS, `/users/${username}/repos`)
}
