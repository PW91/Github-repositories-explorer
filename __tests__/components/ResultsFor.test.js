import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeProvider } from 'emotion-theming'
import { findByTestAtrr } from '>/jest-config/utils'
import theme from '~/styles/theme/theme'
import ResultsFor from '~/components/ResultsFor'

expect.addSnapshotSerializer(serializer)

describe('ResultsFor component', () => {

  it('It renders', () => {
    const wrapper =  shallow(<ResultsFor />)
    expect(findByTestAtrr(wrapper, 'ResultsForView').exists()).toBeTruthy()
  })

  it('Snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <ResultsFor />
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})