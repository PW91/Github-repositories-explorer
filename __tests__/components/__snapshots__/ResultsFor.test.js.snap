// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsFor component Snapshot 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1rem 0;
}

.emotion-6 .emotion-1 {
  margin-right: .5rem;
}

.emotion-0 {
  color: #002442;
  font-size: 1.4rem;
  margin: 0;
}

.emotion-2 {
  color: #002442;
  font-size: 1.6rem;
  margin: 0;
  font-weight: 700;
}

.emotion-4 {
  font-family: 'icofont' !important;
  speak: none;
  font-style: normal;
  font-weight: 400;
  font-variant: normal;
  text-transform: none;
  line-height: 1;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-left: 1rem;
  cursor: pointer;
  font-size: 1.8rem;
  color: #002442;
}

.emotion-4:before {
  content: "\\f102";
}

<div
  className="emotion-6 emotion-7"
  data-test="ResultsForView"
>
  <p
    className="emotion-0 emotion-1"
  >
    resultsFor
  </p>
  <h3
    className="emotion-2 emotion-3"
  />
  <span
    className="emotion-4 emotion-5"
  />
</div>
`;
